src/gpu/intel/jit/codegen/codegen.cpp

```diff
@@ -522,7 +522,7 @@ class ir_to_ngen_t : public ir_visitor_t {
     void fill_buf(const ngen_operand_t &buf_op, int size,
             const ngen_operand_t &pattern = {}) const {
         auto &rd = buf_op.reg_buf_data();
-        type_t type = type_t::u32();
+        type_t type = (pattern.is_invalid() ? type_t::f32() : type_t::u32());
```

Comment:
Would it be reasonable to use different types based on the type of the buffer at the IR level?

Reply:
IR buffers don't have a type so it's not available here. But here we don't really want that. For zero-initialization it's better to use `F` pipe as prologue has a lot of arithmetic operations involving `I` and `L` pipes. So even though it's int8 convolution (and the buffer is used as a dword buffer) using `F` pipe should be more beneficial as we offload some work to the less utilized `F` pipe resulting in less stalls.